src/services/sync-checker.ts

```diff
@@ -159,6 +160,23 @@ export class SyncChecker {
     // Consult Altruist for sync source of truth
     const altruistBlockHeight = await this.getSyncFromAltruist(syncCheckOptions, blockchainSyncBackup)
 
+    // Determine whether altruist is trustworthy or not
+    const isAltruistTrustworthy = altruistBlockHeight >= highestNodeBlockHeight
```

Comment:
shouldn't be lowestNodeBlockHeight? on this case even if the other nodes before the highest one are lower than altruist still makes the altruist not trustworthy